iib/workers/tasks/utils.py

```diff
@@ -240,7 +240,7 @@ def reset_docker_config():
 
 
 @contextmanager
-def set_registry_token(token, container_image):
+def set_registry_token(token, container_image, registry_auths=None):
```

Comment:
Looking at the different conditionals in this function now, would it make sense to have a new `set_registry_auths` function that gets used by `set_registry_token`?

```python
def set_registry_token(token, container_image):
    if not token:
        ...
    if not container_image:
        ...
    registry = ImageName.parse(container_image).registry
    encoded_token = base64.b64encode(token.encode('utf-8')).decode('utf-8')
    registry_auths = {'auths': {registry: encoded_token}}
    with set_registry_auths(registry_auths):
      yield
    return


def set_registry_auths(registry_auths):
    if not registry_auths:
        ...
    docker_config_path = os.path.join(os.path.expanduser('~'), '.docker', 'config.json')
    try:                                                                                            
        log.debug('Removing the Docker config symlink at %s', docker_config_path)                      
        try:                                                                                        
            os.remove(docker_config_path)                                                           
        except FileNotFoundError:                                                                   
            log.debug('The Docker config symlink at %s does not exist', docker_config_path)         
                                                                                                    
        conf = get_worker_config()                                                                  
        if os.path.exists(conf.iib_docker_config_template):                                         
            with open(conf.iib_docker_config_template, 'r') as f:                                   
                docker_config = json.load(f)                                                        
        else:                                                                                       
            docker_config = {}                                                                      
                                                                
        registries = list(registry_auths.get('auths', {}).keys())
        log.debug('Setting the override token for the registries %s in the Docker config', registries)  

        docker_config.setdefault('auths', {})
        docker_config['auths'].update(registry_auths.get('auths', {}))                              
        with open(docker_config_path, 'w') as f:                                                    
            json.dump(docker_config, f)                                                             
                                                                                                    
        yield                                                                                       
    finally:                                                                                        
        reset_docker_config()
```

Beware that I have not tested the code above.

Reply:
Nice! It makes perfect sense